### DISCUSSION TITLE
antialising not working 'generateTexuture' from 'drawRoundedRect'

### DISCUSSION THREAD

**Discussion by leestrument** [pixijs/pixijs#7627] [Q&A]
![무제 2](https://user-images.githubusercontent.com/46497930/126526704-919592e5-c50a-4bf4-b345-c54450096184.png)

Playground : https://www.pixiplayground.com/#/edit/iFYOj3E-6R84DbVh1xzVI

Hi. first sorry for bad english!
I created a texture using app.renderer.generateTexuture(myPIXIGraphics)
I create the graphics using drawRoundedRect
but the rounded corner doesn't looks good.
any help?
Thanks!

Lee

Apple macbook m1 13 2021 / Chrome

**Comment by bigtimebuddy** [comment]
There's a new-ish feature called multisampling (aka MSAA) that might help in these cases. I'm not using `generateTexture`, instead using RenderTexture directly, but it's the same idea.

Instead of this:

```js
const padTexture = app.renderer.generateTexture(padGraphics)
```

Try this:

```js
const bounds = padGraphics.getBounds();
const padTexture = PIXI.RenderTexture.create({
    width: bounds.width,
    height: bounds.height,
});
padTexture.baseTexture.framebuffer.multisample = PIXI.MSAA_QUALITY.MEDIUM;
app.renderer.render(padGraphics, padTexture);
app.renderer.framebuffer.blit();
```

It's a little more cumbersome to write, but produces a smoother result. The next release (6.1.0) will support passing the multisample property to generateTexture.

https://www.pixiplayground.com/#/edit/BN3dy0mAXOWGmXL7qFrtX

**Reply by leestrument** [reply]
Thanks Matt!!!!!
PIXI js team is amazing. I didn't expect I could get the answer in a hour ! wow !!!!!
I hope more straight-forward way could come about converting graphics to texture without needing app variable to get renderer.
for example, 
`const padTexture =  padGraphics.toTexture()`

**Comment by ivanpopelyshev** [comment]
Alternative, you can use https://github.com/pixijs/graphics-smooth , dont need MSAA for that :)

**Reply by ivanpopelyshev** [reply]
but only if you cover this thing with a stroke . We dont have AA for drawRoundedRect fill yet :(

**Reply by bigtimebuddy** [reply]
I was also going to recommend this as an alternative approach too! Here's an example using SmoothGraphics:

https://jsfiddle.net/bigtimebuddy/en9uf03w/

_Note: that this approach requires a stroke, there are limitations currently with fills which @ivanpopelyshev can explain_
